Source/Integrations/NukeIntegration.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  NukeIntegration.swift
+//  AXPhotoViewer
+//
+//  Created by Alessandro Nakamuta on 15/04/18.
+//  Copyright © 2018 Alessandro Nakamuta. All rights reserved.
+//
+import Nuke
+
+class NukeIntegration: NSObject, AXNetworkIntegrationProtocol {
+
+    weak public var delegate: AXNetworkIntegrationDelegate?
+
+    fileprivate var retrieveImageTasks = NSMapTable<AXPhotoProtocol, ImageTask>(keyOptions: .strongMemory, valueOptions: .strongMemory)
+
+    public func loadPhoto(_ photo: AXPhotoProtocol) {
+        if photo.imageData != nil || photo.image != nil {
+            self.delegate?.networkIntegration(self, loadDidFinishWith: photo)
+            return
+        }
+
+        guard let url = photo.url else {
+            return
+        }
+        
+        let progress: ImageTask.ProgressHandler = { [weak self] (_, receivedSize, totalSize) in
+            guard let `self` = self else {
+                return
+            }
+            
+            self.delegate?.networkIntegration?(self, didUpdateLoadingProgress: CGFloat(receivedSize) / CGFloat(totalSize), for: photo)
+        }
+        
+        let completion: ImageTask.Completion = { [weak self] (response, error) in
+            guard let `self` = self else {
+                return
+            }
+            
+            self.retrieveImageTasks.removeObject(forKey: photo)
+            
+            if let imageData = response?.image.animatedImageData {
+                photo.imageData = imageData
+                self.delegate?.networkIntegration(self, loadDidFinishWith: photo)
+            } else if let image = response?.image {
+                photo.image = image
+                self.delegate?.networkIntegration(self, loadDidFinishWith: photo)
+            } else {
+                let error = NSError(
+                    domain: AXNetworkIntegrationErrorDomain,
+                    code: AXNetworkIntegrationFailedToLoadErrorCode,
+                    userInfo: nil
+                )
+                self.delegate?.networkIntegration(self, loadDidFailWith: error, for: photo)
+            }
+        }
+        
+        ImagePipeline.Configuration.isAnimatedImageDataEnabled = true
```

Comment:
We can probably get setting this static variable once in a one-time initializer. Unfortunately there's not a super great way of doing that in Swift (anymore). Perhaps something like this:

```swift
    private static let once: Void = {
        ImagePipeline.Configuration.isAnimatedImageDataEnabled = true
    }()

    override public init() {
        super.init()
        NukeIntegration.once()
    }
```

Reply:
Or we can remove this and left the user decide if he wants to activate Nuke's gif support. We can add this on README. If it is not enabled, it will download the gif as a normal image.